### PR TITLE
chardev: multiple backend support

### PR DESCRIPTION
**Pull request by vivianQizhu**
id: 1588423
Signed-off-by: qizhu <qizhu@redhat.com>

### CODE REVIEW

**Comment by vivianQizhu** [comment]
Depends on: https://github.com/avocado-framework/avocado-vt/pull/1639

File: qemu/tests/cfg/spice.cfg [hunk 1]
```diff
@@ -7,10 +7,10 @@
     display_vm2 = vnc
     vga_vm2 = cirrus
     full_screen = no
-    virtio_ports_virt-tests-vm1 = "vdagent"
-    virtio_port_type_virt-tests-vm1 = "serialport"
-    virtio_port_chardev_virt-tests-vm1 = "spicevmc"
-    virtio_port_name_prefix_virt-tests-vm1 = "com.redhat.spice."
+    serials_virt-tests-vm1 += " vdagent"
```
Comment (yanan-fu) [inline qemu/tests/cfg/spice.cfg]: `serials += " vdagent"`

Comment (vivianQizhu) [reply qemu/tests/cfg/spice.cfg]: @yanan-fu I think that should have `_virt-tests-vm1` as previous defined: `virtio_ports_virt-tests-vm1`, that's for a specific vm, better not change original configuration.

Comment (yanan-fu) [reply qemu/tests/cfg/spice.cfg]: @vivianQizhu  oh~, yes, you are right, i didn't notice there are not only one vm.

File: qemu/tests/cfg/virtio_serial_file_transfer.cfg [hunk 2]
```diff
@@ -3,8 +3,9 @@
     guest_scripts = VirtIoChannel_guest_send_receive.py;windows_support.py
     guest_script_folder = /var/tmp/
     file_transfer_serial_port = vc1
-    virtio_ports = vc1 vc2
-    virtio_port_type = serialport
+    serials += " vc1 vc2"
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_serial_file_transfer.cfg]: `vs` ? (just keep consistent,  they all work)

Comment (vivianQizhu) [reply qemu/tests/cfg/virtio_serial_file_transfer.cfg]: @yanan-fu Let's keep it in this patch, this script does not work anyway, I am rewriting it.

File: qemu/tests/cfg/virtio_console.cfg [hunk 3]
```diff
@@ -22,7 +25,7 @@
                     virtio_console_test = delete_guest_script
 
         # Tests which can run or serialport or console
-        # NOTE: By default VM have booth serialports and consoles. The difference is that it uses serialport or console as tested object. If you want to test pure VM without the other virtio_port devices change virtio_ports (or virtio_port_type) in 'virtserialport' resp 'virtconsole' variants.
+        # NOTE: By default VM have booth serialports and consoles. The difference is that it uses serialport or console as tested object. If you want to test pure VM without the other virtio_port devices change serials (or serial_type) in 'virtserialport' resp 'virtconsole' variants.
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_console.cfg]: `in 'virtserialport_test' resp 'virtconsole_test' variants.`

File: qemu/tests/cfg/virtio_console.cfg [hunk 4]
```diff
@@ -36,43 +39,43 @@
...
                         - rw_host_offline:
                             # console is always connected to underneath virtserialport.
-                            only virtserialport
+                            only virtserialport_test
                             virtio_console_test = rw_host_offline
                         - rw_host_offline_big_data:
                             only Linux
-                            # console is always connected to underneath virtserialport.
-                            only virtserialport
+                            # console is always coffnnected to underneath virtserialport.
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_console.cfg]: typo `coffnnected`

File: qemu/tests/cfg/virtio_port_hotplug.cfg [hunk 5]
```diff
@@ -1,10 +1,9 @@
 - virtio_port_hotplug:
     type = virtio_port_hotplug
-    virtio_ports = vc1 vc2
-    virtio_port_type = serialport
+    serials += " vc1 vc2"
+    serial_type_vc1 = virtserialport
+    serial_type_vc2 = virtconsole
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_port_hotplug.cfg]: Same,  `vc`, `vs`

File: qemu/tests/cfg/virtio_serial_stop_continue.cfg [hunk 6]
```diff
@@ -1,6 +1,6 @@
 - virtio_serial_stop_continue:
-    virtio_ports = vc1
-    virtio_port_type_vc1 = serialport
+    serials += " vc1"
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_serial_stop_continue.cfg]: same

**Comment by vivianQizhu** [comment]
@yanan-fu Updated, expect the name issue, let's focus on the variants name changing in this patch, thanks.

**Comment by yanan-fu** [comment]
Ack

Let's wait,  merge this one together with https://github.com/avocado-framework/avocado-vt/pull/1639

File: qemu/tests/qemu_guest_agent.py [hunk 7]
```diff
@@ -195,13 +195,7 @@ def gagent_create(self, params, vm, *args):
...
         gagent_name = args[1]
 
-        if gagent_serial_type == guest_agent.QemuAgent.SERIAL_TYPE_VIRTIO:
-            filename = vm.get_virtio_port_filename(gagent_name)
-        elif gagent_serial_type == guest_agent.QemuAgent.SERIAL_TYPE_ISA:
-            filename = vm.get_serial_console_filename(gagent_name)
-        else:
-            raise guest_agent.VAgentNotSupportedError("Not supported serial"
-                                                      " type")
+        filename = vm.get_serial_console_filename(gagent_name)
```
Comment (vivianQizhu) [inline qemu/tests/qemu_guest_agent.py]: @yanan-fu Please help review the update, thanks.

Comment (yanan-fu) [reply qemu/tests/qemu_guest_agent.py]: LGTM

Test is running now, will update result later.

Comment (yanan-fu) [reply qemu/tests/qemu_guest_agent.py]: Run gating test loop,  all pass.

**Pull request merged**